source/grid/grid_tools.cc

```diff
@@ -6452,6 +6458,469 @@ namespace GridTools
+          "to equal the number of vertices of the triangulation."));
+
+      // In the case that a marked_vertices vector has been given and none
+      // of its entries is true, we know that this process does not own
+      // any of the incoming points (and it will not send any data) so
+      // that we can take a short cut.
+      const bool has_relevant_vertices =
+        (marked_vertices.size() == 0) ||
+        (std::find(marked_vertices.begin(), marked_vertices.end(), true) !=
+         marked_vertices.end());
+
+      // intersection between two cells:
+      // One rank requests all intersections of owning cell:
+      // owning cell index, cgal vertices of cell
+      using RequestType =
+        std::vector<std::pair<unsigned int, IntersectionRequest>>;
+      // Other ranks send back all found intersections for requesting cell:
+      // requesting cell index, cgal vertices of found intersections
+      using AnswerType =
+        std::vector<std::pair<unsigned int, IntersectionAnswer>>;
+
+      const auto create_request = [&](const unsigned int other_rank) {
+        const auto other_rank_index = translate(other_rank);
+
+        RequestType request;
+        request.reserve(potential_owners_ptrs[other_rank_index + 1] -
+                        potential_owners_ptrs[other_rank_index]);
+
+        for (unsigned int i = potential_owners_ptrs[other_rank_index];
+             i < potential_owners_ptrs[other_rank_index + 1];
+             ++i)
+          request.emplace_back(
+            potential_owners_indices[i],
+            intersection_requests[potential_owners_indices[i]]);
+
+        return request;
+      };
+
+
+      // TODO: this is potentially useful in many cases and it would be nice to
+      // have cache.get_locally_owned_cell_bounding_boxes_rtree(marked_vertices)
+      const auto construct_locally_owned_cell_bounding_boxes_rtree =
+        [&cache](const std::vector<bool> &marked_verts) {
+          const auto cell_marked = [&marked_verts](const auto &cell) {
+            for (const unsigned int v : cell->vertex_indices())
+              if (marked_verts[cell->vertex_index(v)])
+                return true;
+            return false;
+          };
+
+          const auto &boxes_and_cells =
+            cache.get_locally_owned_cell_bounding_boxes_rtree();
+
+          std::vector<
+            std::pair<BoundingBox<spacedim>,
+                      typename dealii::Triangulation<dim>::cell_iterator>>
+            potential_boxes_and_cells;
+
+          for (const auto &box_and_cell : boxes_and_cells)
+            if (cell_marked(box_and_cell.second))
+              potential_boxes_and_cells.emplace_back(box_and_cell);
+
+          return pack_rtree(potential_boxes_and_cells);
+        };
+
+
+      const auto answer_request =
+        [&](const unsigned int &other_rank,
+            const RequestType & request) -> AnswerType {
+        AnswerType answer;
```

Comment:
https://github.com/dealii/dealii/pull/15166#discussion_r1186855537

Reply:
> In the point code, the size of answer is the same as of `request` and we answer only the number of found cells for each point. Here we only send back relevant ones. Alternatively, one could send back `std::vectors` of intersection and their size would indicate the number of intersections. But we can keep it like this

I would like to keep it this way. Fund intersections for a given `request`, can be determined using `std::vector<unsigned int> recv_ptrs`. This is also why I would like to keep this member.